File: algorithms/0766.toeplitz-matrix/solution.go
func isToeplitzMatrix(matrix [][]int) bool {
	rows := len(matrix)
	cols := len(matrix[0])

	for m := 1; m < rows; m++ {
		for n := 1; n < cols; n++ {
			if matrix[m][n] != matrix[m-1][n-1] {
				return false
			}
		}
	}

	return true
}

File: algorithms/0766.toeplitz-matrix/solution.rs
impl Solution {
    pub fn is_toeplitz_matrix(matrix: Vec<Vec<i32>>) -> bool {
        for m in 1..matrix.len() {
            for n in 1..matrix[0].len() {
                if matrix[m][n] != matrix[m - 1][n - 1] {
                    return false;
                }
            }
        }

        true
    }
}
